src/main/java/net/fabricmc/loom/configuration/accesswidener/AccessWidenerJarProcessor.java

```diff
@@ -76,41 +78,52 @@ public String getId() {
 
 	@Override
 	public void setup() {
-		LoomGradleExtension loomGradleExtension = LoomGradleExtension.get(project);
-		File awPath = loomGradleExtension.getAccessWidenerPath().get().getAsFile();
-
-		if (!awPath.exists()) {
-			throw new RuntimeException("Could not find access widener file @ " + awPath.getAbsolutePath());
+		LoomGradleExtension extension = LoomGradleExtension.get(project);
+		Path awPath = extension.getAccessWidenerPath().get().getAsFile().toPath();
+
+		// Read our own mod's access widener, used later for producing a version remapped to intermediary
+		try {
+			 modAccessWidener = Files.readAllBytes(awPath);
+	 	} catch (NoSuchFileException e) {
+			throw new RuntimeException("Could not find access widener file @ " + awPath.toAbsolutePath());
+		} catch (IOException e) {
+			throw new RuntimeException("Failed to read access widener: " + awPath);
 		}
 
-		inputHash = Checksum.sha256(awPath);
+		// Write a collated full access widener for hashing it, and forward it to the access widener we'll apply
+		var fullWriter = new AccessWidenerWriter();
+		var forwardingVisitor = new ForwardingVisitor(fullWriter, accessWidener);
 
-		try (BufferedReader reader = new BufferedReader(new FileReader(awPath))) {
-			accessWidenerReader.read(reader);
-		} catch (IOException e) {
-			throw new RuntimeException("Failed to read project access widener file");
-		}
+		try (var remappingProvider = new RemappingProvider(project)) {
+			var remappingVisitor = new RemappingDecorator(forwardingVisitor, remappingProvider, "named");
 
-		//Remap accessWidener if its not named, allows for AE's to be written in intermediary
-		if (!accessWidener.getNamespace().equals("named")) {
-			try {
-				List<String> validNamespaces = loomGradleExtension.getMappingsProvider().getMappings().getMetadata().getNamespaces();
+			// For our own mod, we include local AWs, and enable strict parsing
+			var ownModReader = new AccessWidenerReader(remappingVisitor);
+			ownModReader.setStrictMode(true);
+			ownModReader.read(modAccessWidener);
 
-				if (!validNamespaces.contains(accessWidener.getNamespace())) {
-					throw new UnsupportedOperationException(String.format("Access Widener namespace '%s' is not a valid namespace, it must be one of: '%s'", accessWidener.getNamespace(), String.join(", ", validNamespaces)));
+			// For other mods, we do not use strict mode, and also filter for global AWs
+			var globalReader = new AccessWidenerReader(new GlobalOnlyDecorator(remappingVisitor));
+			for (var entry : Constants.MOD_COMPILE_ENTRIES) {
```

Comment:
Please dont use `var`

Reply:
Alright, I thought I could because someone else already snuck some past you :-P

Reply:
Yeah, could have even been me tbh. Its very hard to read when they are spammed like this. Going to add a checkstyle rule to prevent them. 👍